### ISSUE TITLE
Canvas JSON tool configuration

### ISSUE THREAD

**Issue by michaelwheeler** [academic-innovation/django-lti#1]
We can provide a view or utility to help applications generate a [JSON tool configuration appropriate for Canvas](https://canvas.instructure.com/doc/api/file.lti_dev_key_config.html#configuring-the-tool-in-canvas).

**Comment by cmurtaugh** [comment]
Very interested in this! The LTI 1.3 installation process is a little tedious and providing the Canvas JSON config is a great help. We might be able to derive a JSON schema from [the Canvas code](https://github.com/instructure/canvas-lms/blob/ac948b6aff5078c19ea465cf8ecc659a82f5d959/lib/schemas/lti/tool_configuration.rb).

**Comment by michaelwheeler** [comment]
Apparently [LTI Dynamic Registration support is in active development for Canvas](https://youtu.be/BKqB1RvzGIM?t=1655), so it probable makes sense to prioritize #3 and then reevaluate whether this still adds any value for Canvas specifically.

**Comment by nikolas** [comment]
@cmurtaugh I have a draft PR for this here: https://github.com/academic-innovation/django-lti/pull/135

Although Dynamic Registration seems to be very convenient, the JSON config is a pretty reliable, simple way to do this. It's also pretty simple to put together this view within the application though, so I'm on the fence on whether or not this should be included in django-lti. If nothing else, maybe we can provide documentation with steps on how to create this view, until Dynamic Registration is more widely supported.

**Comment by cmurtaugh** [comment]
Yeah; we've experimented a bit with Dynamic Registration in a few of our tools, and while it makes the initial setup process very easy, it limits the changes you can make via the UI later on (say, if you want to add another custom variable, or change a placement or something). We've ended up having to delete and re-install tools a few times in order to change some settings. IIRC the spec might have something to say about the re-registration/update process, but I'm not sure that Canvas supports it yet (and I haven't figured out how to implement it in a tool, in any case).
